alicloud/monitoring_metric.go

```diff
@@ -157,7 +157,8 @@ func listCMMetricStatistics(ctx context.Context, d *plugin.QueryData, granularit
 	var results []map[string]interface{}
 	err = json.Unmarshal([]byte(stats.Datapoints), &results)
 	if err != nil {
-		return nil, err
+		plugin.Logger(ctx).Error("Invalid JSON response", err)
+		return nil, nil
```

Comment:
@ParthaI This fix looks a little risky to me, why are we ignoring all errors in the `json.Unmarshal` call? Wouldn't we do better to understand why the API response is `nil` sometimes, and handle that better before we attempt to call `json.Unmarshal`?